pkg/decoder/decoder.go

```diff
@@ -23,27 +25,47 @@ func NewPayload(content []byte, offset int64) *Payload {
 	return &Payload{content, offset}
 }
 
-// Decoder splits raw data based on `\n`, and sends those messages to a channel
+// Decoder splits raw data into messages using '\n' for single-line logs and re for multi-line logs
+// and sends them to a channel
 type Decoder struct {
 	InputChan  chan *Payload
 	OutputChan chan message.Message
+	countDownDuration time.Duration
```

Comment:
I would put that as a constant

Reply:
Agreed but it's interesting to inject it for unit-tests